### ISSUE TITLE
Graph Panel widget: only reset axes when needed

### ISSUE THREAD

**Issue by s-fong** [opencor/opencor#1777]
In a graph panel containing a selection of ticked and unticked plots after a simulation was run to completion:
- x axis was custom set to a given range in one of the graph panels
- choose another graph to view in the same panel, either by clicking its name on the legend, or by checking the box under the Graphs dialog on the left. 

Second step resets the axis to the original default range, instead of the custom range set in the first step!

**Comment by agarny** [comment]
It's true that whenever you show a graph, the axes of its corresponding graph panel get reset (so that we can see all of its graphs that are visible can be seen). However, I appreciate your point, but keep in mind that it's not the only case where axes will get reset in that manner. From what I can tell, axes get reset when (in square brackets what I think we might want to do when it comes to resetting or not axes):
 - some simulation properties are changed [NO RESET];
 - a graph is added [NO RESET];
 - a graph is removed [NO RESET];
 - a graph is shown (i.e your case) [NO RESET];
 - the GUI needs to be updated (e.g. a file is reloaded) [RESET]; and
 - we update our simulation results [RESET].

So, in which cases should we actually reset axes, if at all? Any views @s-fong, @nickerso, @mirams, @ssafaei? (Anyone else?)

(FWIW, to implement each of the above cases is only a matter of changing one parameter value in a function call, so it shouldn't take more than a few minutes and then making sure that everything works as expected.)

**Comment by nickerso** [comment]
My suggestion would be that if an axis is set to "autorange" then the axis should be re-scaled any time the displayed data changes, to show the full range. If an axes has had its range set by the user (or the SED-ML?) then it should never be reset.

**Comment by agarny** [comment]
Agreed, that makes more sense. Ok, so before resetting axes, I am going to check whether they are set to "autorange" and, if so, reset them otherwise I will leave as such.

**Comment by agarny** [comment]
Ok, I think we want what I mentioned in https://github.com/opencor/opencor/issues/1777#issuecomment-415313079, keeping in mind what @nickerso mentioned in https://github.com/opencor/opencor/issues/1777#issuecomment-415378143.

Thus, if we for example add/remove/show/hide a graph, then the axes will only be reset if their range was the same as the one needed to plot all the graphs before adding/removing/showing/hiding the aforementioned graph. If the axes' range was different then they will remain the same.

When running a model, the axes get updated no matter what and as needed so that all of the (selected) graphs are visible.

**Comment by s-fong** [comment]
With the 23-Aug snapshot:  works well when the user has defined custom axes.
But it appears that auto scaling doesn't happen at all when custom axes were not set, and the user is showing or removing graphs from a given panel after the simulation is run.

**Comment by nickerso** [comment]
agreed, doesn't seem to be rescaling the axes when new graphs are added or made visible when no custom axis has been defined. An example to demonstrate...

1. Open https://models.physiomeproject.org/workspace/25d/rawfile/e3ef4cf57b9be0b9f5ccb72e98dac46f2381add5/Noble_1962.cellml
2. Go to simulation panel and set max step to 0.1
3. Run
4. Plot the `Nai` variable, axes is not rescaled and you get the following:
![nai-1](https://user-images.githubusercontent.com/811244/44557493-c0697280-a792-11e8-989f-276ae0914cbc.PNG)
5. Clear the simulation results and reset the model
6. Run the simulation
7. Axis rescaled and correctly displays as this
![nai-2](https://user-images.githubusercontent.com/811244/44557518-daa35080-a792-11e8-944e-906135601f93.PNG)
8. Plot the `V` variable, axis not rescaled and you get the following (with V not visible)
![nai-vm-1](https://user-images.githubusercontent.com/811244/44557549-0d4d4900-a793-11e8-9fef-14793152a392.PNG)
9. Clear the simulation results and reset the model, run it again and the axes are correctly rescaled
![nai-vm-2](https://user-images.githubusercontent.com/811244/44557566-35d54300-a793-11e8-8def-4950ff49b3c2.PNG)
10. Hide the `V` graph and the axis is not rescaled
![nai-vm-3](https://user-images.githubusercontent.com/811244/44557577-438ac880-a793-11e8-8c04-518f1c9607c5.PNG)

**Comment by agarny** [comment]
Argh, thanks @s-fong and @nickerso, I can indeed confirm the regression. I am going to look into it.
